### PR TITLE
Feat: fine tune OpenSSH for jenkins agents

### PR DESCRIPTION
**Pull request by dduportal**
This PR introduces 2 new features on *all* the jenkins-agents images:

* Predefined OpenSSH authorized keys from https://github.com/jenkins-infra/aws/blob/main/ec2_agents_authorized_keys to allow EC2 use cases (which requires OpenSSH keypair on EC2) for faster agent startups
* Add a script on each OS to dynamically add new public key during the boot sequence (cloud-init based most of the time):

Linux example for AWS:

```yaml
runcmd:
  # Add the instance key to the jenkins user
  - /usr/local/bin/add_auth_key_to_user.sh "$(curl -sSL http://169.254.169.254/latest/meta-data/public-keys/0/openssh-key)" jenkins
```

Windows example for AWS:

```powershell
<powershell>
$instanceKey = Invoke-RestMethod -uri 'http://169.254.169.254/latest/meta-data/public-keys/0/openssh-key'
C:\addSSHPubKey.ps1 "${instanceKey}" "Administrator"
</powershell>
```

### CODE REVIEW

**Review by timja — APPROVED** [review]

File: Jenkinsfile [hunk 1]
```diff
@@ -75,9 +75,9 @@ pipeline {
             }
           }
           stage('Build') {
-            when {
-              branch 'main'
-            }
+            // when {
```
[the remark this answers is not in the document]
Comment (dduportal) [reply Jenkinsfile:78]: Good catch: end of day deserves better attention when I open PRs ;)

**Comment by dduportal** [comment]
Just tested and the Windows image does not accept SSH authentication: need to debug.

File: Jenkinsfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (dduportal) [reply Jenkinsfile]: Done

**Comment by dduportal** [comment]
@timja The last issue was the URL given for retrieveing the `authorized_keys` file was not pointing to the "raw" file in GitHub: the file on the filesystem had a bunch of HTML instead of the expected keys. Fixed, along with your review's feedback, rebase, and merged as it's minor change.

**Pull request merged**